File: Blackoffice/Models/Mas_Product_Model.vb
Imports MD
Public Class Mas_Product_Model
    Public Class value_SCR
        Public Respon As New ResponseModels
        Public Data As List(Of Product_SCR)
    End Class
    Public Class value_List
        Public Respon As New ResponseModels
        Public Data As List(Of Product_List)
    End Class
    Public Class value_data
        Public Respon As New ResponseModels
        Public Data As New Product_data
    End Class

    Public Class Product_data

        Public Product As Product = New Product()
        Public PRODUCT_SED_UNIT As List(Of MAS_PRODUCT_SED_UNIT) = New List(Of MAS_PRODUCT_SED_UNIT)

    End Class
    Public Class Product_List
        Private _ID As String
        Public Property ID() As String
            Get
                Return _ID
            End Get
            Set(value As String)
                _ID = value
            End Set
        End Property

        Private _PCODE As String
        Public Property PCODE() As String
            Get
                Return _PCODE
            End Get
            Set(value As String)
                _PCODE = value
            End Set
        End Property

        Private _PNAME As String
        Public Property PNAME() As String
            Get
                Return _PNAME
            End Get
            Set(value As String)
                _PNAME = value
            End Set
        End Property
        Private _PTNAME As String
        Public Property PTNAME() As String
            Get
                Return _PTNAME
            End Get
            Set(value As String)
                _PTNAME = value
            End Set
        End Property
        Private _PGNAME As String
        Public Property PGNAME() As String
            Get
                Return _PGNAME
            End Get
            Set(value As String)
                _PGNAME = value
            End Set
        End Property
        Private _ACTIVEFLAG As String
        Public Property ACTIVEFLAG() As String
            Get
                Return _ACTIVEFLAG
            End Get
            Set(value As String)
                _ACTIVEFLAG = value
            End Set
        End Property
        Private _MDATE As DateTime
        Public Property MDATE() As DateTime
            Get
                Return _MDATE
            End Get
            Set(value As DateTime)
                _MDATE = value
            End Set
        End Property
    End Class

    Public Class Product_SCR
        Private _PCODE As String
        Public Property PCODE() As String
            Get
                Return _PCODE
            End Get
            Set(value As String)
                _PCODE = value
            End Set
        End Property
        Private _PCODE2 As String
        Public Property PCODE2() As String
            Get
                Return _PCODE2
            End Get
            Set(value As String)
                _PCODE2 = value
            End Set
        End Property
        Private _PNAME As String
        Public Property PNAME() As String
            Get
                Return _PNAME
            End Get
            Set(value As String)
                _PNAME = value
            End Set
        End Property
        Private _PGNAME As String
        Public Property PGNAME() As String
            Get
                Return _PGNAME
            End Get
            Set(value As String)
                _PGNAME = value
            End Set
        End Property
        Private _PTNAME As String
        Public Property PTNAME() As String
            Get
                Return _PTNAME
            End Get
            Set(value As String)
                _PTNAME = value
            End Set
        End Property
        Private _COST_STD As String
        Public Property COST_STD() As Double
            Get
                Return _COST_STD
            End Get
            Set(value As Double)
                _COST_STD = value
            End Set
        End Property

        Private _UNITCODE As String
        Public Property UNITCODE() As String
            Get
                Return _UNITCODE
            End Get
            Set(value As String)
                _UNITCODE = value
            End Set
        End Property
        Private _UNITNAME As String
        Public Property UNITNAME() As String
            Get
                Return _UNITNAME
            End Get
            Set(value As String)
                _UNITNAME = value
            End Set
        End Property
    End Class
    Public Class Product
        Public Property ID As Integer
        Public Property PCODE As String
        Public Property PCODE2 As String
        Public Property PNAME As String
        Public Property PGCODE As String
        Public Property PGNAME As String

        Public Property UNITCODE As String
        Public Property IMGNAME1 As String
        Public Property COST_STD As Double

        Public Property ACTIVEFLAG As String
        Public Property DESCRIPTION As String
        Public Property REMARK As String
        Public Property [CDATE] As DateTime
        Public Property CUSE As String
        Public Property MDATE As DateTime
        Public Property MUSE As String
        Public Property ISTAT As String


        Public Property PTCODE As String
        Public Property PRICE_STD As Double

    End Class
    Public Class MAS_PRODUCT_SED_UNIT
        Public Property SEQ As Integer
        Public Property PCODE As String
        Public Property UNITCODE As String
        Public Property UNITNAME As String
        Public Property QTY_MAIN As Integer
        Public Property QTY_SED As Integer

    End Class
    Public Class TMP_IMP_PRODUCT

        Public Property SEQ As String
        Public Property RUN_ID As String
        Public Property PCODE As String
        Public Property PNAME As String
        Public Property DESCRIPTION As String
        Public Property IMG_NAME As String
        Public Property PTCODE As String
        Public Property PTNAME As String
        Public Property PGCODE As String
        Public Property PGNAME As String
        Public Property PGSCODE As String
        Public Property PGSNAME As String
        Public Property BRANDCODE As String
        Public Property BRANDNAME As String
        Public Property MODEL As String
        Public Property COLORCODE As String
        Public Property COLORNAME As String
        Public Property UNITCODE As String
        Public Property UNITNAME As String
        Public Property PACKCODE As String
        Public Property PACKNAME As String
        Public Property AGE As String
        Public Property NUM_OWD As String
        Public Property DEPCT As String
        Public Property PRICESTD As String
        Public Property WEIGHT As String
        Public Property Remark As String
        Public Property CUSE As String
    End Class

End Class

File: TouchPOS_API/Models/Mas_Product_Model.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using BaseRepository.Helpers;
using TouchPOS_API.Helpers;
using System.Configuration;

namespace TouchPOS_API.Models
{
    public class Mas_Product_Model
        
    {
        public class Product_data
        {
            public Product Product = new Product();
            public List<MAS_PRODUCT_SED_UNIT> PRODUCT_SED_UNIT = new List<MAS_PRODUCT_SED_UNIT>();

        }
        public class Product_SCR
        {

            [Column]
            public string PCODE { get; set; }
 
            [Column]
            public string PNAME { get; set; }
            [Column]
            public string PTNAME { get; set; }
            [Column]
            public string PGNAME { get; set; }
            [Column]
            public double  COST_STD { get; set; }
            [Column]
            public string UNITCODE { get; set; }

            [Column]
            public string UNITNAME { get; set; }
            
        }
        public class Product_List
        {
            [Column]
            public string ID { get; set; }
            [Column]
            public string PCODE { get; set; }

            [Column]
            public string PNAME { get; set; }           
            [Column]
            public string PGNAME { get; set; }
            [Column]
            public string PTNAME { get; set; }
            [Column]
            public DateTime MDATE { get; set; }
            [Column]
            public string ACTIVEFLAG { get; set; }
        }



        [Table("MAS_PRODUCT")]
        public class Product
        {
            [Column]
            public int ID { get; set; }
            [Column]
            public string PCODE { get; set; }
            [Column]
            public string PCODE2 { get; set; }
            [Column]
            public string PNAME { get; set; }
            [Column]
            public string PGCODE { get; set; }
            [Column]
            public string PGNAME { get; set; }
            [Column]
            public string UNITCODE { get; set; }
            [Column]
            public string IMGNAME1 { get; set; }
            [Column]
            public double COST_STD { get; set; }
            [Column]
            public string ACTIVEFLAG { get; set; }
            [Column]
            public string DESCRIPTION { get; set; }
            [Column]
            public string REMARK { get; set; }
            [Column]
            [Date(Now = true)]
            public DateTime CDATE { get; set; }
            [Column]
            public string CUSE { get; set; }
            [Column]
            [Date(Now =true)]
            public DateTime MDATE { get; set; }
            [Column]
            public string MUSE { get; set; }
            [Column]
            public string ISTAT { get; set; }
            [Column]
            public double PRICE_STD { get; set; }

            [Column]
            public string PTCODE { get; set; }
        }

        [Table("MAS_PRODUCT_SED_UNIT")]
        public class MAS_PRODUCT_SED_UNIT
        {
            [Column]
            public string SEQ { get; set; }
            [Column]
            public string PCODE { get; set; }
            [Column]
            public string UNITCODE { get; set; }
            [Column]
            public string UNITNAME { get; set; }
            [Column]
            public int QTY_MAIN { get; set; }
            [Column]
            public int QTY_SED { get; set; }

        }
        [Table("TMP_IMP_PRODUCT")]
        public class TMP_IMP_PRODUCT
        {
            [Column]
            public string SEQ { get; set; }
            [Column]
            public string RUN_ID { get; set; }
            [Column]
            public string PCODE { get; set; }
            [Column]
            public string PNAME { get; set; }
            [Column]
            public string DESCRIPTION { get; set; }
            [Column]
            public string IMG_NAME { get; set; }
            [Column]
            public string PTCODE { get; set; }
            [Column]
            public string PTNAME { get; set; }
            [Column]
            public string PGCODE { get; set; }
            [Column]
            public string PGNAME { get; set; }
            [Column]
            public string PGSCODE { get; set; }
            [Column]
            public string PGSNAME { get; set; }
            [Column]
            public string BRANDCODE { get; set; }
            [Column]
            public string BRANDNAME { get; set; }
            [Column]
            public string MODEL { get; set; }
            [Column]
            public string COLORCODE { get; set; }
            [Column]
            public string COLORNAME { get; set; }
            [Column]
            public string UNITCODE { get; set; }
            [Column]
            public string UNITNAME { get; set; }
            [Column]
            public string PACKCODE { get; set; }
            [Column]
            public string PACKNAME { get; set; }
            [Column]
            public string AGE { get; set; }
            [Column]
            public string NUM_OWD { get; set; }
            [Column]
            public string DEPCT { get; set; }
            [Column]
            public string PRICESTD { get; set; }
            [Column]
            public string WEIGHT { get; set; }
            [Column]
            public string Remark { get; set; }
            [Column]
            public string CUSE { get; set; }
            [Column]
            [Date(Now = true)]
            public string CDATE { get; set; }
        }
        [Table("SHOPBUTTONS")]
        public class SHOPBUTTONS
        {

            [Column]
            public int SHOPID { get; set; }
            [Column]
            public string POSID { get; set; }
            [Column]
            public int BTNID { get; set; }
            [Column]
            public string BTNGROUP { get; set; }
            [Column]
            public string BTNNAME { get; set; }
            [Column]
            public double  BTNPRICE { get; set; }
            [Column]
            [Date(Now = true)]
            public DateTime LUPDATE { get; set; }
            [Column]
            public string barcode { get; set; }
            [Column]
            public string flagSt { get; set; }
            [Column]
            public string pic1 { get; set; }
            [Column]
            public string pic2 { get; set; }
            [Column]
            public double  cost { get; set; }
            [Column]
            public string MenuTypeID { get; set; }
            [Column]
            public int flg { get; set; }

        }




    }



}